ExploreASL_Initialize.m

```diff
@@ -697,111 +639,242 @@
             end
-            if      x.S.Sets1_2Sample(iSet)==1
-                    fprintf(', codes for paired data');
-            elseif  x.S.Sets1_2Sample(iSet)==2
-                    fprintf(', codes for two-sample data');
-            elseif  x.S.Sets1_2Sample(iSet)==3
-                    fprintf([', continuous variate (with ' num2str(length(unique(x.S.SetsID(:,iSet)))) ' unique values)']);
-            end                    
-
-            fprintf('\n');
         end
-else    
-    fprintf('%s\n','No sets are defined');
-end
+    end
+
+    if ~isempty(regexp(x.subject_regexp, '^(\^|)\.\*(\$|)$'))
+        warning('Subject regexp not specific! Check that no wrong folders are included as subjects');
+    end
+
+    fprintf('\n');
 
-if ~isfield(x,'M0')
-%     warning('M0 option missing!');
-else
-    fprintf('\n%s\n',['M0 option selected is "' num2str(x.M0) '"']);
 end
 
-fprintf('%s\n',['x.DELETETEMP = ' num2str(x.DELETETEMP) ' (delete temporary files)']);
-fprintf('%s\n',['x.Quality    = ' num2str(x.Quality) ' (0 = fast try-out; 1 = normal high quality)']);
+%% -----------------------------------------------------------------------
+%% Add ExploreASL Directory
+function addExploreASLDirectory(MyPath)
 
+    % Define paths (should be equal when loading data or only initializing)
+    addpath(MyPath); % ExploreASL
+    subfolders_to_add = {'Functions', 'mex', 'Modules', 'Testing',...
+        fullfile('Modules', 'SubModule_Structural'), ...
+        fullfile('Modules', 'SubModule_ASL'), ...
+        fullfile('Modules', 'SubModule_Population'), ...
+        'Development', 'External', ...
+        fullfile('External','isnear'), ...
+        fullfile('External','DCMTK'), ...
+        fullfile('Testing', 'UnitTests'), ...
+        fullfile('External','ExploreQC'), ...
+        fullfile('External','SPMmodified'), ...
+        fullfile('External','SPMmodified','matlabbatch'),...
+        fullfile('External','SPMmodified','xASL'),...
+        fullfile('External','SPMmodified','toolbox','cat12'), ...
+        fullfile('External','SPMmodified','toolbox','LST'), ...
+        fullfile('External','SPMmodified','toolbox','OldNorm') ...
+        genpath(fullfile('External','bids-matlab'))};
+    % Iterate over subfolders which should be added
+    for ii=1:length(subfolders_to_add)
+        addpath(fullfile(MyPath,subfolders_to_add{ii}));
+    end
+    
+end
 
 %% -----------------------------------------------------------------------
-%% 3) Print warnings
-fprintf('\n%s\n\n','---------------------------------------------');
+%% Define input parser
+function p = inputParsing(varargin)
 
-field_symbol = {'subject_regexp'};
+    % Initialize input parser
+    p = inputParser;
+    
+    % Define valid input variables
+    validDataParPath = @(variable) ischar(variable) || isempty(variable);
+    validImportArray = @(variable) ischar(variable) || isempty(variable) || isnumeric(variable);
```

Comment:
Why do all these numerical/logical/Boolean input values allow `ischar(var)`?

Reply:
The compiled version give us char arrays, don't you remember? :)